lib/logstash/plugin_mixins/event_support/event_factory_adapter/fallback_impl.rb

```diff
@@ -0,0 +1,65 @@
+# encoding: utf-8
+require 'jruby'
+
+module LogStash::PluginMixins::EventSupport::EventFactoryAdapter::FallbackImpl
+
+  # The event_factory method is effectively final and should not be re-defined by plugins.
+  #
+  # @return an event factory object with a `new_event(Hash)` API
+  def event_factory
+    @_event_factory || JRuby.reference(self).synchronized do
+      @_event_factory ||= create_event_factory
+    end
+  end
+
+  # The `targeted_event_factory` method is effectively final and should not be re-defined.
+  # If the plugin defines a `target => ...` option than this method will return a factory
+  # that respects the set target, otherwise (no target) returns {#event_factory}.
+  #
+  # @return an event factory object with a `new_event(Hash)` API
+  def targeted_event_factory
+    @_targeted_event_factory ||= begin # not synchronized - fine to initialize twice
+                                   raise ArgumentError.new('config(:target) not present') unless respond_to?(:target)
+                                   target.nil? ? event_factory : TargetedEventFactory.new(event_factory, target)
+                                 end
+  end
+
+  private
+
+  # @api private
+  # @since LS 7.14
+  def create_event_factory
+    BasicEventFactory::INSTANCE
+  end
```

Comment:
If we take my other suggestion to simplify `FallbackImpl#event_factory`, we won't need this:

```suggestion
```